spec/support/fake_analytics.rb

```diff
@@ -47,7 +47,8 @@ def track_event(event, original_attributes = {})
         when Hash
           value.each do |key, val|
             current_keypath = keypath + [key]
-            if pii_attr_names.include?(key) && !pii_like_keypaths.include?(current_keypath)
+            if pii_attr_names.include?(key) &&
+               pii_like_keypaths.none? { |keypath| current_keypath.take(keypath.count) == keypath }
```

Comment:
This change allows for allowlisting a whole tree of paths. I'm not _sure_ we want this instead of forcing a developer to specify every individual full path. I was encountering errors originally, though unsure now if it was related to this logic or more relevant to the change below.

Reply:
I feel like the spirit of the check is to list every single keypath, since ideally there wouldn't be too many? I don't feel that strongly

Reply:
When thinking about this change, I had these in mind as some more extreme examples of where we're doing a lot of work to exempt PII keypaths:

https://github.com/18F/identity-idp/blob/0919778754a4686730a6058730c0b101e7d4cc91/app/forms/idv/doc_pii_form.rb#L49-L58

https://github.com/18F/identity-idp/blob/0919778754a4686730a6058730c0b101e7d4cc91/spec/support/doc_pii_helper.rb#L2-L14

Reply:
I'll check what was failing here, I think the revision to use a hash added an extra level to the keypath where we'd need to start exempting each key of the new hash.

Reply:
Ok, I reverted to the original logic in 3946ad2bf.

As seen in the commit, this does require a lot of additional work to accommodate the new nested key. However, I think this is less a problem with the PII detector, and more a problem that we don't give a meaningful "type" to the type of error, and instead just reuse the field name, causing it to get double-flagged for both the field and the error type.